codesigndocuitests/codesigngroup.go

```diff
@@ -0,0 +1,424 @@
+			if len(profileOptions) == 1 {
+				selectedProfileOption = profileOptions[0]
+
+				fmt.Printf("Provisioning Profile to sign target (%s): %s\n", bundleID, selectedProfileOption)
+			} else {
+				sort.Strings(profileOptions)
+
+				fmt.Println()
+				question := fmt.Sprintf("Select the Provisioning Profile to sign target with bundle ID: %s", bundleID)
+				selectedProfileOption, err = goinp.SelectFromStringsWithDefault(question, 1, profileOptions)
+				if err != nil {
+					return nil, fmt.Errorf("failed to read input: %s", err)
+				}
+			}
+
+			for _, profile := range profiles {
+				option := fmt.Sprintf("%s (%s)", profile.Name, profile.UUID)
+				if option == selectedProfileOption {
+					selectedBundleIDProfileMap[bundleID] = profile
+				}
+			}
+		}
+		if len(selectedBundleIDProfileMap) != len(bundleIDProfilesMap) {
+			return nil, fmt.Errorf("failed to find Provisioning Profiles for ipa export")
+		}
+
+		var collectedCodeSignGroup export.CodeSignGroup
+
+		collectedCodeSignGroup = export.NewIOSGroup(*selectedCertificate, selectedBundleIDProfileMap)
+
+		fmt.Println()
+		log.Infof("Codesign settings will be used for %s method:", codesignMethod(collectedCodeSignGroup))
+		printCodesignGroup(collectedCodeSignGroup)
+
+		collectedCodeSignGroups = append(collectedCodeSignGroups, collectedCodeSignGroup)
+
+		fmt.Println()
+		question := fmt.Sprintf("Do you want to collect other code sign files for (%s)", testRunnerID)
+		question += "\n(select NO to finish collecting codesign files and continue)"
+		anotherExport, err := goinp.AskForBoolWithDefault(question, false)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read input: %s", err)
+		}
+		if !anotherExport {
+			break
+		}
+	}
+
+	return collectedCodeSignGroups, nil
+}
+
+// collectExportSelectableCodeSignGroups returns every possible codesigngroup which can be used to export an ipa file
+func collectExportSelectableCodeSignGroups(testRunner IOSTestRunner, installedCertificates []certificateutil.CertificateInfoModel, installedProfiles []profileutil.ProvisioningProfileInfoModel) []export.SelectableCodeSignGroup {
+	bundleIDEntitlemenstMap := map[string]plistutil.PlistData{}
+	bundleIDEntitlemenstMap = testRunner.BundleIDEntitlementsMap()
+
+	bundleIDs := []string{}
+	for bundleID := range bundleIDEntitlemenstMap {
+		bundleIDs = append(bundleIDs, bundleID)
+	}
+	codeSignGroups := export.CreateSelectableCodeSignGroups(installedCertificates, installedProfiles, bundleIDs)
+
+	log.Debugf("Codesign Groups:")
+	for _, group := range codeSignGroups {
+		log.Debugf(group.String())
+	}
+
+	if len(codeSignGroups) == 0 {
+		return []export.SelectableCodeSignGroup{}
+	}
+
+	codeSignGroups = export.FilterSelectableCodeSignGroups(codeSignGroups,
+		export.CreateEntitlementsSelectableCodeSignGroupFilter(bundleIDEntitlemenstMap),
+	)
+
+	fmt.Println()
+	// Handle if UITest target used NON xcode managed profile
+	if len(codeSignGroups) > 0 && !testRunner.IsXcodeManaged() {
```

Comment:
at this point len(codeSignGroups) is always > 0, restructure this, like:

```
if testRunner.IsXcodeManaged() {

} else {
}
```

Reply:
👍